File: funnyvote/app/src/main/java/com/heaton/funnyvote/data/promotion/PromotionDataSource.java
package com.heaton.funnyvote.data.promotion;

import com.heaton.funnyvote.database.Promotion;
import com.heaton.funnyvote.database.User;

import java.util.List;

import rx.Observable;

public interface PromotionDataSource {
    Observable<List<Promotion>> getPromotionList(User user);

    void savePromotionList(List<Promotion> promotionList);
}

File: funnyvote/app/src/main/java/com/heaton/funnyvote/data/promotion/PromotionDataSource.kt
package com.heaton.funnyvote.data.promotion

import com.heaton.funnyvote.database.Promotion
import com.heaton.funnyvote.database.User
import com.heaton.funnyvote.database.VoteData

interface PromotionDataSource {

    interface GetPromotionsCallback {
        fun onPromotionsLoaded(promotionList: List<Promotion>)

        fun onPromotionsNotAvailable()
    }

    fun getPromotionList(user: User, callback: GetPromotionsCallback)

    fun savePromotionList(promotionList: List<Promotion>)
}
